### PR TITLE
Add docs to explain important functions in ONNX Infra

### PR DESCRIPTION
**Pull request by houseroad**
Target to cover major frequently used functions

### CODE REVIEW

File: docs/ImportantPythonFunctions.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import onnx
+import os
+from onnx import numpy_helper
+
+# Preprocessing: create a Numpy array
+numpy_array = numpy.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]], dtype=float)
+print('Original Numpy array:\n{}\n'.format(numpy_array))
+
+# Convert the Numpy array to a TensorProto
+tensor = numpy_helper.from_array(numpy_array)
```
Comment (bddppq) [inline docs/ImportantPythonFunctions.md]: `from_array` takes an optional argument `name`.

File: docs/ImportantPythonFunctions.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+Runnable IPython notebooks:
+- [load_model.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/load_model.ipynb)
+
+## Saving an ONNX Model
+```python
+import onnx
+
+onnx_model = ... # Your model in memory
+
+# Save the ONNX model
```
Comment (bddppq) [inline docs/ImportantPythonFunctions.md]: Can be done in a separate diff: We should wrap this into a helper function (say `save_model`), especially soon we will have support for storing tensor values in separate files.

File: docs/ImportantPythonFunctions.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+# Convert the Numpy array to a TensorProto
+tensor = numpy_helper.from_array(numpy_array)
+print('TensorProto:\n{}'.format(tensor))
+
+# Convert the TensorProto to a Numpy array
+new_array = numpy_helper.to_array(tensor)
+print('After round trip, Numpy array:\n{}\n'.format(numpy_array))
+
+# Save the TensorProto
+with open(os.path.join(os.path.dirname('tensor.pb'), 'wb') as f:
```
Comment (bddppq) [inline docs/ImportantPythonFunctions.md]: There should be typo here.

**Comment by bddppq** [comment]
Actually, since you have setup notebooks, should we simply not put any python code in the markdown file (since they are not verified by tests)?

File: docs/ImportantPythonFunctions.md (hunk 1 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: Sure, will cover this case. :-)

File: docs/ImportantPythonFunctions.md (hunk 2 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: Good point. Working on it, will rebase this diff on that one.

**Comment by houseroad** [comment]
After some thoughts, I think keeping short(er) version in the markdown file makes sense. The summarized version is easy to check, and users can quickly grab the usage from just one page. :-)

File: docs/ImportantPythonFunctions.md (hunk 3 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: Will fix it.

File: docs/ImportantPythonFunctions.md (hunk 2 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: https://github.com/onnx/onnx/pull/692

File: docs/ImportantPythonFunctions.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+# The protobuf definition can be found here:
+# https://github.com/onnx/onnx/blob/master/onnx/onnx.proto
+
+
+# Create one input (ValueInfoProto)
+X = helper.make_tensor_value_info('X', TensorProto.FLOAT, [1, 2])
+
+# Create one output (ValueInfoProto)
+Y = helper.make_tensor_value_info('X', TensorProto.FLOAT, [1, 2])
```
Comment (dzhulgakov) [inline docs/ImportantPythonFunctions.md]: typo in the name X -> Y

File: docs/ImportantPythonFunctions.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+- [make_model.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/make_model.ipynb)
+- [Protobufs.ipynb](https://github.com/onnx/onnx/tree/master/onnx/examples/Protobufs.ipynb)
+
+## Checking an ONNX Model
+```python
+import onnx
+import os
+
+# Preprocessing: load the ONNX model
+model_path = os.path.join(os.path.dirname('model.onnx')
```
Comment (dzhulgakov) [inline docs/ImportantPythonFunctions.md]: unfinished line? is this code runnable? :)

File: docs/ImportantPythonFunctions.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+original_model_str = original_model.SerializeToString()
+
+print('The model before optimization:\n{}'.format(original_model))
+
+# A full list of supported optimization passes can be found here:
+# https://github.com/onnx/onnx/blob/master/onnx/optimizer.py#L21
+pass_list = ['fuse_consecutive_transposes']
+
+# Apply the optimization on the original serialized model
+optimized_model = onnx.load_from_string(optimizer.optimize(original_model_str, pass_list))
```
Comment (dzhulgakov) [inline docs/ImportantPythonFunctions.md]: should we have a wrapper function that hides to/from string? and also there should be a "no-brainer" pass list as default

File: docs/ImportantPythonFunctions.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+original_model = helper.make_model(graph, producer_name='onnx-examples')
+
+# Check the model and print Y's shape information
+onnx.checker.check_model(original_model)
+print('Before shape inference, the shape info of Y is:\n{}'.format(original_model.graph.value_info))
+
+# Apply shape inference on the model
+original_model_str = original_model.SerializeToString()
+inferred_model_str = shape_inference.infer_shapes(original_model_str)
+inferred_model = onnx.load_from_string(inferred_model_str)
```
Comment (dzhulgakov) [inline docs/ImportantPythonFunctions.md]: same here - hide serialization inside the library

File: docs/ImportantPythonFunctions.md (hunk 4 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: Good catch!

File: docs/ImportantPythonFunctions.md (hunk 5 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: Yep, forgot to delete something.

File: docs/ImportantPythonFunctions.md (hunk 7 above)
Comment (houseroad) [reply docs/ImportantPythonFunctions.md]: Yep, I will do that in a different PR. And then rebase the introduction doc.

**Comment by houseroad** [comment]
@onnxbot retest this please

**Pull request merged**